dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/CrudControllerAdvice.java

```diff
@@ -146,6 +146,22 @@ protected void initBinder(WebDataBinder binder) {
     binder.registerCustomEditor(UID.class, new UIDParamEditor());
   }
 
+  @ExceptionHandler
+  @ResponseBody
+  public WebMessage queryRuntimeException(QueryRuntimeException ex) {
```

Comment:
We could potentially encapsulate this code somewhere else.
Let's discuss the possibilities.